addons/website_event_track/static/src/js/website_event_track.js

```diff
@@ -96,3 +96,19 @@ publicWidget.registry.websiteEventTrack = publicWidget.Widget.extend({
         }
     },
 });
+
+publicWidget.registry.eventTrackForm = publicWidget.Widget.extend({
+    selector: '.o_wevent_event_search_box_track',
+    events: {
+        'input .search-query': '_onClearSearchInput',
```

Comment:
Use `search` event here, so no need to add if condition in function. Give it a try.